zulipterminal/core.py

```diff
@@ -142,14 +142,17 @@ def stream_muting_confirmation_popup(self, button: Any) -> None:
         self.loop.widget = PopUpConfirmationView(self, question,
                                                  mute_this_stream)
 
-    def narrow_to_stream(self, button: Any) -> None:
-        already_narrowed = self.model.set_narrow(stream=button.stream_name)
+    def _narrow_to(self, button: Any, **narrow: Any) -> None:
+        already_narrowed = self.model.set_narrow(**narrow)
         if already_narrowed:
             return
 
         self.model.found_newest = False
+
         # store the steam id in the model (required for get_message_ids...)
-        self.model.stream_id = button.stream_id
+        if hasattr(button, 'stream_id'):  # FIXME Include in set_narrow?
```

Comment:
I think it's okay, since we will have to set it in two functions anyway.

Reply:
I left this note for now, as it honestly feels like if we set a narrow then we should be able to do this there, rather than handling it separately.